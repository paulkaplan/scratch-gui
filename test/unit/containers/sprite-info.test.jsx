/* eslint-env jest */
import React from 'react'; // eslint-disable-line no-unused-vars
import {mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux'; // eslint-disable-line no-unused-vars
import renderer from 'react-test-renderer';

import SpriteInfo from '../../../src/containers/sprite-info'; // eslint-disable-line no-unused-vars

describe('SpriteSelectorItem Container', () => {
    const mockStore = configureStore();
    let vm;
    let targets;
    let store;
    const getContainer = function () {
        return (
            <Provider store={store}>
                <SpriteInfo
                    x={100}
                    y={-100}
                    direction={50}
                    name={'name'}
                />
            </Provider>
        );
    };

    beforeEach(() => {
        vm = {
            postSpriteInfo: jest.fn(),
            renameSprite: jest.fn()
        };
        targets = {
            editingTarget: 'TARGET'
        };
        store = mockStore({vm, targets});
    });

    test('matches the snapshot', () => {
        const component = renderer.create(getContainer());
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should use the vm#postSpriteInfo callback for x input', () => {
        const wrapper = mount(getContainer());
        wrapper.find('input[placeholder="x"]').simulate('change', {target: {value: '123'}});
        wrapper.find('input[placeholder="x"]').simulate('keypress', {key: 'Enter'});
        expect(vm.postSpriteInfo).toBeCalledWith({x: 123});
    });

    test('should use the vm#postSpriteInfo callback for y input', () => {
        const wrapper = mount(getContainer());
        wrapper.find('input[placeholder="y"]').simulate('change', {target: {value: '5'}});
        wrapper.find('input[placeholder="y"]').simulate('blur');
        expect(vm.postSpriteInfo).toBeCalledWith({y: 5});
    });

    test('should use the vm#renameSprite callback for name', () => {
        const input = mount(getContainer()).find('input[placeholder="Name"]');
        input.simulate('change', {target: {value: 'cat cat'}});
        input.simulate('blur');
        expect(vm.renameSprite).toBeCalledWith('TARGET', 'cat cat');
    });
});
