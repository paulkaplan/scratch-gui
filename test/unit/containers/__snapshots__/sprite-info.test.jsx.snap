// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpriteSelectorItem Container matches the snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className=""
  >
    <div
      className={undefined}
    >
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          Sprite
        </span>
        <input
          className=""
          disabled={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          placeholder="Name"
          tabIndex="1"
          type="text"
          value="name"
        />
      </label>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <img
          className=""
          src="test-file-stub"
        />
      </div>
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          x
        </span>
        <input
          className="undefined"
          disabled={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          placeholder="x"
          tabIndex="2"
          type="text"
          value={100}
        />
      </label>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <img
          className=""
          src="test-file-stub"
        />
      </div>
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          y
        </span>
        <input
          className="undefined"
          disabled={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          placeholder="y"
          tabIndex="3"
          type="text"
          value={-100}
        />
      </label>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className={undefined}
    >
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          Show
        </span>
        <div>
          <div
            className=""
            onClick={[Function]}
            tabIndex="4"
          >
            <img
              className={undefined}
              src="test-file-stub"
            />
          </div>
          <div
            className=""
            onClick={[Function]}
            tabIndex="4"
          >
            <img
              className={undefined}
              src="test-file-stub"
            />
          </div>
        </div>
      </label>
    </div>
    <div
      className={undefined}
    >
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          Direction
        </span>
        <input
          className="undefined"
          disabled={undefined}
          label="Direction"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          tabIndex="5"
          type="text"
          value={50}
        />
      </label>
    </div>
    <div
      className={undefined}
    >
      <label
        className={undefined}
      >
        <span
          className={undefined}
        >
          Rotation
        </span>
        <select
          className=""
          disabled={undefined}
          onChange={[Function]}
          value={undefined}
        >
          <option
            value="left-right"
          >
            left-right
          </option>
          <option
            value="don't rotate"
          >
            don't rotate
          </option>
          <option
            value="all around"
          >
            all around
          </option>
        </select>
      </label>
    </div>
  </div>
</div>
`;
